src/librustc_back/rpath.rs

```diff
@@ -40,10 +40,7 @@ pub fn get_rpath_flags<F, G>(config: RPathConfig<F, G>) -> Vec<String> where
     debug!("preparing the RPATH!");
 
     let libs = config.used_crates.clone();
-    let libs = libs.into_iter().filter_map(|(_, l)| {
-        l.map(|p| p.clone())
-    }).collect::<Vec<_>>();
-
+    let libs = libs.into_iter().filter_map(|(_, l)| l).collect::<Vec<_>>();
```

Comment:
Yeah, redundant it seems, if I remember, `cloned` wasn't implemented for `l` due to being an Option with an owned T. Is it a requirement to change the typehint in this PR?